js/selectors/dashboard-selectors.js

```diff
@@ -44,14 +45,52 @@ export const getStudentProgress = createSelector(
   }
 )
 
-// Returns sorted students.
+// Returns a mapping from student IDs to their total progress across all activities
+export const getStudentTotalProgress = createSelector(
+  [ getStudents, getStudentProgress ],
+  (students, studentProgress) => {
+    return students.map(student => {
+      const id = student.get('id').toString()
+      return studentProgress.get(id).toList().reduce((sum, progress) => sum + progress)
+    })
+  }
+)
+
+// A comparison function to sort students by last and then first name
+const compareStudentsByName = (student1, student2) => {
```

Comment:
I made a comment on your PR about sorting - this is the solution I had in mind. I ended up wanting it to break ties for students with the same progress, let me know what you think

Reply:
It was a good catch, thanks. You can check my fix, as it's a bit simpler. It's just enough to concatenate last name and first name with whitespace in between. JS will do all the work for us.

Reply:
Kirk's comment:
> To sort names correctly in the general case, first names and last names should be compared independently. Either can contain spaces, for instance.

So, I'm wrong above. Your approach is better.

Reply:
Got it! I did throw a `.toLocaleLowerCase()` in there so it handles capitalization like yours does